### DISCUSSION TITLE
Some emails stuck in Pending state after Postal V1 to V2 migration

### DISCUSSION THREAD

**Discussion by MrRichThomas** [postalserver/postal#2150] [Help with using Postal] [answered]
Hi, I wondered if I could get some advice. We've just migrated from Postal V1 to V2, following the 'Installing on a new server with existing data' instructions here: https://docs.postalserver.io/install/upgrade-to-v2#installing-on-a-new-server-with-existing-data

We didn't realise that we needed to run 'postal upgrade' prior to starting the server after restoring the databases. This meant that some of the frontend functionality wasn't working. We then ran 'postal upgrade' but we didn't stop Postal first. The upgrade ran successfully, but during the upgrade we were still receiving mail from our applications, and I think it's this mail that is now stuck in a Pending state.

If possible I need to get Postal to attempt delivery of some of the mail, and some of the mail I need to delete.

Do you know how I might do that? At the moment if I click on any of the Pending emails in Postal I don't have the option to deliver / release, it just says "Message is currently being processed."

This is what one of the Pending messages currently looks like in the database.

id: 1442694
token: t8sfitTFieGH
scope: outgoing
rcpt_to: ####@####.co.uk
mail_from: no-reply@######.co.uk
subject: ##########
message_id: 9319e6b7-49dc-4bdd-bd5a-08ac6ff78524@rp.mail.######.co.uk
timestamp: 1662545496.624512
route_id: NULL
domain_id: 32
credential_id: 17
status: Pending
held: 0
size: 14628
last_delivery_attempt: NULL
raw_table: raw-2022-09-07
raw_body_id: 284
raw_headers_id: 283
 inspected: 0
spam: 0
spam_score: 0.00
threat: 0
threat_details: NULL
bounce: 0
bounce_for_id: 0
tag: NULL
loaded: NULL
clicked: NULL
received_with_ssl: 1
hold_expiry: NULL
tracked_links: 0
tracked_images: 0
parsed: 0
endpoint_id: NULL
endpoint_type: NULL

Any help would be much appreciated, thanks.

**Comment by willpower232** [comment, marked as answer]
I think someone added a PR for the missing documentation so will try to get that sorted.

I think you need to clear the queue fields in the database for the affected messages in order for them to be picked up again by the workers.

**Reply by MrRichThomas** [reply]
Thanks @willpower232 yeah it was my colleague who added the PR for the docs. :)

Sorry, which table would I find the queue fields in?

Thanks
Rich

**Reply by willpower232** [reply]
Just had to go digging to find it `Theres a table called queued_messages. Just make locked_by, locked_at, retry_after to NULL`

**Reply by MrRichThomas** [reply]
Ah, it's in the main Postal server, I was thinking it would be somewhere in each individual mail server DB. Awesome, thanks very much for your help Will!
